parser/src/main/java/org/hibernate/hql/ast/spi/SingleEntityQueryRendererDelegate.java

```diff
@@ -288,7 +430,19 @@ public void sortSpecification(String collateName, String orderSpec) {
     * @param collateName optional collation name
     * @param isAscending sort direction
     */
-	protected abstract void addSortField(PropertyPath propertyPath, String collateName, boolean isAscending);
```

Comment:
Implementing sorting and grouping should be optional but should not require users to provide any implementation not even a blank method if they do not wish to support it. This change is backward compatible.